clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -972,16 +975,28 @@ private Record nextFetchedRecord() {
         private List<ConsumerRecord<K, V>> fetchRecords(int maxRecords) {
             if (isFetched)
                 return Collections.emptyList();
+            if (nextInlineException != null) {
```

Comment:
@hachikuji I did ask you to review this patch. But you asked for something that is out of the scope of this patch. Changing that behavior takes definitely more than 5 minutes and more importantly, I did not like that behavior (at least before we have the transaction semantics that was committed after I opened the review request).

If this patch address a problem and does not add any concern, why can't we merge it? Isn't it allowed to address independent concern separately?